### ISSUE TITLE
Whitelisting doesn't work (for me)

### ISSUE THREAD

**Issue by gbrylinski** [rollbar/rollbar-gem#793]
### Before #792 

Before this patch, whitelisting was working properly except situation when `config.filter_parameters` was set for Rails application. `config.filter_parameters` is supposed to list fields that are *blacklisted* and should be excluded/scrubbed from logs, reports, etc. Pre #792 code adds content of `config.filter_parameters` to blacklist as well as whitelist what can lead to passing sensitive data into report.

so, for configuration like this:

```ruby
# application.rb
config.filter_parameters = [:password, :api_key]

# initializers/rollbar.rb
config.scrub_whitelist = true
config.scrub_fields = [:user_id, :name, :surname]
```

I would expect that only fields `[:user_id, :name, :surname]` keep real values and any other field is scrubbed. But, because `[:password, :api_key]` are added to the list also `password` and `api_key` values are whitelisted and get reported.

### After #792 

it seems to not work. I started with following config:

```ruby
# initializers/rollbar.rb
config.scrub_whitelist = [:password]
```

I would expect all fields except `password` to be scrubbed but all values are preserved. For:

```ruby
# initializers/rollbar.rb
config.scrub_whitelist = [:password]
config.scrub_fields = []
```

again, all except `password` should get scrubbed but again all are preserved. When I use:

```ruby
# application.rb
config.filter_parameters = [:utf8, :commit]

# initializers/rollbar.rb
config.scrub_whitelist = [:password]
config.scrub_fields = [:username]
```

I get `utf8`, `commit` & `username` scrubbed and everything else is whitelisted. It seems like whitelist is applied to blacklist and then blacklisted fields get scrubbed. My understanding of whitelisting is slightly different - we let pass only things that are on the whitelist and anything else get scrubbed. That's also what documentation says:

```
scrub_whitelist

Set to true to switch scrubbing mode to whitelist. In whitelist mode, everything but the fields provided in scrub_fields gets scrubbed. The values will be replaced with a "*" before being sent.
```

### Pull request #794

I've made a PR based on master before #792 was applied. It does following things:

- changes specs so they test whitelist behaviour instead of blacklist
- changes `Url` & `Params` scrubbers
- makes necessary changes to `RequestDataExtractor` to support whitelist in urls
- properly ignores `filter_parameters` when whitelisting is used

**Comment by rokob** [comment]
What you are describing is the `scrub_all` option. The `scrub_whitelist` indeed does not work correctly with this option as I just realized. `scrub_whitelist` is the list of things not to scrub, but it does not imply to scrub everything but what is on that list. It is intended as a means of excluding from scrubbing things that are already set in `scrub_fields` most often by including some framework pre-defined keys.

**Comment by rokob** [comment]
I'm going to make the whitelist work with scrub_all before the next release. To get the behaviour you are describing you would set scrub_all to true, then specify the scrub_whitelist to have the keys you do not want scrubbed. If scrub_all is not set to true, then scrub_whitelist will be keys that do not get scrubbed even if they are in scrub_fields.

**Comment by gbrylinski** [comment]
Thanks @rokob, use of `scrub_all` + `scrub_whitelist` makes perfect sense. I have no idea how often `scrub_all` is used but mutually exclusive white/black lists feel more common and simpler to understand. Intuitively, to scrub all fields I would simply use empty white list.

**Comment by jessewgibbs** [comment]
@ArturMoczulski  @rokob should we still consider this PR or was this already addressed in 2.18.2?

**Comment by ArturMoczulski** [comment]
I believe this has been addressed in #795 and the changes are included in `v2.18.2`
